### DISCUSSION TITLE
How to read a xlsx file synchronously?

### DISCUSSION THREAD

**Discussion by raahed** [exceljs/exceljs#1680] [Q&A] [answered]
I'm trying to read a excel file into a csv-like looking array [ArrayLine1,ArrayLine2,..] with exceljs in nodejs. Exceljs has probably no solution to read a file synchronuously, so I have to build a solution around that. But all my attempts with async functions around the exceljs code doesn't work with the synchronuous frontend.
```js

const workbook = new Excel.Workbook()

var raw = []
var line
workbook.xlsx.readFile(filename).then(() => {

    workbook.eachSheet((sheet, sid) => {

       sheet.eachRow((row, rid) => {

           line = []
           row.eachCell((cell, cid) => {
               
               line.push(cell.value)
           })
           
           raw.push(line)
       })
    })
})

//use raw / write the raw to another file
console.log(raw) //echo's: []
```
Is there any solution to read files syncrronuously?

**Comment by skypesky** [comment, marked as answer]
@raahed Maybe you can use async / await .
